### DISCUSSION TITLE
infb data--splitobject error

### DISCUSSION THREAD

**Discussion by GreyRockIQ** [satijalab/seurat#5433] [Q&A] [answered]
Hello
I'm stuck early errors in integrated analysis..

ifnb.list <- SplitObject(ifnb, split.by = "stim")

Error in FetchData(object = object, vars = split.by) : 
None of the requested variables were found: stim

 Any suggestions?

Thanks
GreyRock

**Comment by samuel-marsh** [comment]
Hi @GreyRockIQ,

Can you please post the full code you are running?

Thank you,
Sam

**Comment by GreyRockIQ** [comment]
Hi @samuel-marsh 
Thanks for response.
Here is the code and error

ifnb.list <- Seurat::SplitObject(ifnb, split.by = "stim")

Error in FetchData(object = object, vars = split.by) : 
None of the requested variables were found: stim

Thanks
GreyRock

**Comment by samuel-marsh** [comment]
Hi @GreyRockIQ 

can you post the entire code?  From loading packages to the point the error occurs?

Thanks,
Sam

**Comment by GreyRockIQ** [comment]
Hi @samuel-marsh 
Here is the full code

```{r}
LoadData("ifnb")
```
Warning in if (is.na(desc)) { :
  the condition has length > 1 and only the first element will be used
Warning in if (is.na(desc)) { :
  the condition has length > 1 and only the first element will be used
An object of class Seurat 
14053 features across 13999 samples within 1 assay 
Active assay: RNA (14053 features, 0 variable features)


```{r}
ifnb
```
An object of class Seurat 
13714 features across 2690 samples within 1 assay 
Active assay: RNA (13714 features, 2000 variable features)

```{r}
ifnb.list <- Seurat::SplitObject(ifnb, split.by = "stim")
```
Error in FetchData(object = object, vars = split.by) : 
None of the requested variables were found: stim

```{r}
ifnb.list <- lapply(X = ifnb.list, FUN = function(x) {
    x <- Seurat::NormalizeData(x) 
    x <- Seurat::FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})
```
Error in lapply(X = ifnb.list, FUN = function(x) { : 
  object 'ifnb.list' not found

```{r}
features <  Seurat::SelectIntegrationFeatures(object.list = ifnb.list)
```
rror in lapply(X = X, FUN = FUN, ...) : object 'ifnb.list' not found

Thanks
GreyRock

**Comment by samuel-marsh** [comment, marked as answer]
Unfortunately I can't replicate the error and it works just fine in my instance.  If you remove the ifnb.SeuratData package and then re-install that dataset does it solve the issue.  

Could you please post the output of `sessionInfo` if the error persists?

Thanks,
Sam

**Reply by Ibisanmi1** [reply]
please i am facing the same error, i try remove and reinstall. same error keep occuring 


ifnb.list <- SplitObject(ifnb, split.by = "stim") Error in slot(object = object, name = s) : no slot of name "images" for this object of class "Seurat"

**Reply by znxfdd** [reply]
data.updated = UpdateSeuratObject(object = ifnb)
ifnb.list <- SplitObject(data.updated, split.by = "stim")

**Comment by GreyRockIQ** [comment]
Hi Sam
Downloading the data folder and loading data file into R solved the issue.
Thanks
GreyRock

**Reply by Ibisanmi1** [reply]
please how did you download the data folder
